### PR TITLE
MM-33445 new endpoint `/sharedchannels/{team_id}`

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary
This PR provides a new endpoint for fetching all shared channels in a team with support for pagination.

API Docs:  https://github.com/mattermost/mattermost-api-reference/pull/622

**To be merged into feature branch.**

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-33445

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: api4/shared_channel.go [hunk 1]
```diff
@@ -6,12 +6,45 @@ package api4
...
 }
 
+func getSharedChannels(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireTeamId()
+	if c.Err != nil {
+		return
+	}
+
+	// make sure remote cluster service is enabled.
+	if _, appErr := c.App.GetRemoteClusterService(); appErr != nil {
```
Comment (nronas) [inline api4/shared_channel.go:25]: Awesome, i've seen that we merged #33444 which makes this returns a 501. 👍

**Review by nronas — APPROVED** [review]
This LGTM

File: api4/shared_channel.go [hunk 2]
```diff
@@ -6,12 +6,45 @@ package api4
...
+
+	"github.com/mattermost/mattermost-server/v5/model"
 )
 
 func (api *API) InitSharedChannels() {
+	api.BaseRoutes.SharedChannels.Handle("/{team_id:[A-Za-z0-9]+}", api.ApiSessionRequired(getSharedChannels)).Methods("GET")
 	api.BaseRoutes.SharedChannels.Handle("/remote_info/{remote_id:[A-Za-z0-9]+}", api.ApiSessionRequired(getRemoteClusterInfo)).Methods("GET")
 }
 
+func getSharedChannels(c *Context, w http.ResponseWriter, r *http.Request) {
```
Comment (mgdelacroix) [inline api4/shared_channel.go:18]: does it make sense to add this endpoint to local mode?

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+1:

File: api4/shared_channel.go (hunk 2 above)
Comment (wiggin77) [reply api4/shared_channel.go:18]: I can't think of a reason not to.  I added a ticket for that:  https://mattermost.atlassian.net/browse/MM-33838

**Pull request merged**